### PR TITLE
nerd-font-patcher: 2.2.2 -> 3.0.1

### PR DESCRIPTION
**Pull request by idlip**
###### Description of changes

Changelog: https://github.com/ryanoasis/nerd-fonts/releases/tag/v3.0.0
The changelog is for font, so i guess applies for patcher too.
As far as i can tell, from 2.2.2 there is addition of whole bunch of material icons.
`2.2.2` had total of 5k icons, `3.0.0` has 11k+ icons. 

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by idlip** [comment]
The package builds successfully with `nix-build -E 'with import <nixpkgs> {}; callPackage ./default.nix {}'`, but i got this error while running the package

```python
Nerd Fonts Patcher v3.0.0 (4.1.1) (ff 20230101)
Traceback (most recent call last):
  File "/nix/store/8zxmwbjkbb4s5f217ll4lkql0pw7zn5z-nerd-font-patcher-3.0.0/bin/.nerd-font-patcher-wrapped", line 1996, in <module>
    main()
  File "/nix/store/8zxmwbjkbb4s5f217ll4lkql0pw7zn5z-nerd-font-patcher-3.0.0/bin/.nerd-font-patcher-wrapped", line 1951, in main
    args = setup_arguments()
  File "/nix/store/8zxmwbjkbb4s5f217ll4lkql0pw7zn5z-nerd-font-patcher-3.0.0/bin/.nerd-font-patcher-wrapped", line 1859, in setup_arguments
    logger.critical("FontnameParser module missing (bin/scripts/name_parser/Fontname*), specify --makegroups 0")
NameError: name 'logger' is not defined
```

Which i assume is wrong, as the python file does have `import logging` line.

I assume that this happens because I only ran `nix-build`, but if I git clone and run the script manually it does not give this error.

I suspect this wont happen if i install it in shell or on system.

Please do check and suggest me if there is something wrong, or for changes to be made!

**Comment by metiulekm** [comment]
I feel like the culprit is this:
1. The new version uses another module here <https://github.com/ryanoasis/nerd-fonts/blob/04e42c1c69b8d74fbbd55c126a2ee6f42d5cbbd9/font-patcher#L41>. This module is not available in our package. Hence, `FontnameParserOK` is set to false.
2. We go into the argument setup here <https://github.com/ryanoasis/nerd-fonts/blob/04e42c1c69b8d74fbbd55c126a2ee6f42d5cbbd9/font-patcher#L1950>. Note that `logger` is not available yet.
3. We get here: <https://github.com/ryanoasis/nerd-fonts/blob/1be89753e122894d584e4746826dca7b3f39a406/font-patcher#L1880>. `FontnameParserOK` is false, and the default value of the other variable is 1, so we get inside and call the `logger`, which fails.

Now that would be a fatal error regardless, so the right solution is probably to make `bin/scripts/name_parser` available. The fact that `logger` is called is an upstream bug, but hopefully it's not that important.

The `name_parser` module had also been imported in 2.2.2, but there the defaults were different and did not use it.

**Comment by idlip** [comment]
@metiulekm Thank you for going through this, tbh I'm not a cs student nor i know any programming language, so I couldn't dig that.

Sorry that i didnt notice it earlier that, it searches for FontnamePatcher.py and FontnameTools.py which are supplied in bin/scripts folder.

You explanation made me realize that nerdfonts already provide stable `FontPatcher.zip`, and it includes just the file needed, so i just extracted and during patch unzip it, and viola everything works as expected.
We need those two files, and we get it from zip.

I have tested it on fonts, it patches as expected with every glyphs!

I'm not sure if its the best design, or nix-way to fetch and unzip... But it works!

Please do suggest if there is a better way! If not, ig its ready to go in!

File: pkgs/applications/misc/nerd-font-patcher/default.nix [hunk 1]
```diff
@@ -1,36 +1,38 @@
-{ python3Packages, lib, fetchFromGitHub }:
+{ python3Packages, lib, unzip, fetchFromGitHub }:
 
 python3Packages.buildPythonApplication rec {
   pname = "nerd-font-patcher";
-  version = "2.2.2";
+  version = "3.0.0";
```
Comment (ck3d) [inline pkgs/applications/misc/nerd-font-patcher/default.nix:5]: Could you please update to version 3.0.1 from yesterday

File: pkgs/applications/misc/nerd-font-patcher/default.nix [hunk 2]
```diff
@@ -1,36 +1,38 @@
...
 
   # This uses a sparse checkout because the repo is >2GB without it
   src = fetchFromGitHub {
     owner = "ryanoasis";
     repo = "nerd-fonts";
     rev = "v${version}";
     sparseCheckout = [
-      "font-patcher"
-      "/src/glyphs"
+      "FontPatcher.zip"
```
Comment (ck3d) [inline pkgs/applications/misc/nerd-font-patcher/default.nix:13]: There is a separate release package. We can now avoid sparseCheckout und hand written unzip by using fetchzip with following path:
https://github.com/ryanoasis/nerd-fonts/releases/download/v3.0.1/FontPatcher.zip

File: pkgs/applications/misc/nerd-font-patcher/default.nix [hunk 3]
```diff
@@ -1,36 +1,38 @@
...
     ];
-    sha256 = "sha256-boZUd1PM8puc9BTgOwCJpkfk6VMdXLsIyp+fQmW/ZqI=";
+    sha256 = "sha256-7pI2t9SHv8y8pEVHqbIECt2I9HGdgEKwWFLM3/DCK8g=";
   };
 
   propagatedBuildInputs = with python3Packages; [ fontforge ];
 
   format = "other";
+  nativeBuildInputs = [ unzip ];
```
Comment (ck3d) [inline pkgs/applications/misc/nerd-font-patcher/default.nix:21]: not needed when fetchzip is used

File: pkgs/applications/misc/nerd-font-patcher/default.nix [hunk 4]
```diff
@@ -1,36 +1,38 @@
...
 
   postPatch = ''
-    sed -i font-patcher \
+    unzip FontPatcher.zip -d nerdpatcher
```
Comment (ck3d) [inline pkgs/applications/misc/nerd-font-patcher/default.nix:24]: not needed when fetchzip is used

File: pkgs/applications/misc/nerd-font-patcher/default.nix [hunk 5]
```diff
@@ -1,36 +1,38 @@
...
       -e 's,__dir__ + "/src,"'$out'/share/nerd-font-patcher,'
   '';
 
   dontBuild = true;
 
   installPhase = ''
-    mkdir -p $out/bin $out/share/nerd-font-patcher
-    install -Dm755 font-patcher $out/bin/nerd-font-patcher
-    cp -ra src/glyphs $out/share/nerd-font-patcher
+    mkdir -p $out/bin $out/share/nerd-font-patcher $out/bin
```
Comment (ck3d) [inline pkgs/applications/misc/nerd-font-patcher/default.nix:32]:
```suggestion
    mkdir -p $out/bin $out/share/nerd-font-patcher
```

File: pkgs/applications/misc/nerd-font-patcher/default.nix [hunk 6]
```diff
@@ -1,36 +1,38 @@
...
+    install -Dm755 nerdpatcher/font-patcher $out/bin/nerd-font-patcher
+    cp -ra nerdpatcher/src/glyphs $out/share/nerd-font-patcher
+    cp -ra nerdpatcher/bin $out/bin/bin
```
Comment (ck3d) [inline pkgs/applications/misc/nerd-font-patcher/default.nix:35]: the bin directory contains a script directory with Python classes. Please copy them to a lib directory.

**Review by ck3d — CHANGES_REQUESTED** [review]

**Comment by idlip** [comment]
Hey @ck3d Thank you for these suggestions, I have made the changes.

Hopefully the file tree is be fine.
Is it fine, i have added myself as maintainer?
You can say no, i can remove myself, no problem with me.

**Comment by ck3d** [comment]
Result of `nixpkgs-review pr 231301` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>nerd-font-patcher</li>
  </ul>
</details>

**Comment by ck3d** [comment]
Thanks for appling my suggestions. Still one issue: fetchzip failes because the ZIP has a flat list of files. Option stripRoot=false could help.

**Comment by idlip** [comment]
@ck3d Ah, I knew it, I removed that line and was testing.

Fixed it now!

**Comment by ck3d** [comment]
Result of `nixpkgs-review pr 231301` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>nerd-font-patcher</li>
  </ul>
</details>

**Review by ck3d — APPROVED** [review]
Thanks, LGTM. Executable runs.

**Pull request merged**